packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol

```diff
@@ -191,24 +238,39 @@ contract Executor is IExecutor {
     return (success, returnData);
   }
 
-  function _handleFailure(
-    bool isNative,
-    bool hasIncreased,
+  /**
+   * @notice Sends funds to the specified recovery address
+   * @dev Called if the external call data fails, it's not a contract, or the amount in native
+   * asset is incorrect.
+   * @param _isNative - Whether the asset is native or not
+   * @param _hasIncreased - Whether the allowance was increased
+   * @param _assetId - Asset associated with call
+   * @param _to - Where call was attempted
+   * @param _recovery - Where to send funds
+   * @param _amount - Amount to send
+   */
+  function _sendToRecovery(
+    bool _isNative,
+    bool _hasIncreased,
     address _assetId,
     address payable _to,
     address payable _recovery,
     uint256 _amount
   ) private {
-    if (!isNative) {
+    if (_amount == 0) {
```

Comment:
what happens if we got here with `msg.value != _args.amount` ??

If the amount is 0 but the msg.value has some extra (somehow, idk how), then that msg.value will get stuck here.